kube-controllers/cmd/kube-controllers/main.go

```diff
@@ -121,14 +121,38 @@ func main() {
 	// Create the context.
 	ctx, cancel := context.WithCancel(context.Background())
 
+	// Create the status file. We will only update it if we have healthchecks enabled.
+	s := status.New(statusFile)
+
 	log.Info("Ensuring Calico datastore is initialized")
-	initCtx, cancelInit := context.WithTimeout(ctx, 10*time.Second)
-	defer cancelInit()
-	err = calicoClient.EnsureInitialized(initCtx, "", "k8s")
 	if err != nil {
 		log.WithError(err).Fatal("Failed to initialize Calico datastore")
 	}
 
+	s.SetReady("StartupCalicoDatastoreInitialization", false, "initialized to false")
+	initCtx, cancelInit := context.WithTimeout(ctx, 60*time.Second)
+	// Loop until context expires or calicoClient is initialized.
```

Comment:
We should call `defer cancelInit()` here rather than calling it directly later after the loop?

Reply:
If i defer it then it won't happen until main returns. Calling it after the loop means the context will be canceled as soon as we're done with it. The only exit from the loop is to break (which means the cancelInit will be called) or a "fatal" exit which means the process exits in which case we don't care if we call the cancel.

Reply:
Right. That's fine with me.